opendatadiscovery/odd-platform
# Usage report implementation

**What changes did you make?** (Give an overview)

This PR introduces new functionality: usage report. This is an information, which is shown on the main page and contains the following blocks:

1. Data entities total count
2. Data entities count splitted by data entity class
3. Unfilled entities count

Unfilled entities count is data entities count, to which user didn't make any manual updates (tags, owners, etc.)
If user makes one of the following updates data entity is counted as filled:

- business name
- description
- tags
- owners
- metadata
- custom group
- terms
- dataset field description
- dataset field label
- dataset field enum
- manually created data entity group

If user fills one of these fields and then remove them data entity again is counted as unfilled

closes #661 

**How to test**

Need to fill data entity with internal information, check, that unfilled count is decremented, remove all info and check, that count is incremented again
